core/trino-main/src/main/java/io/trino/operator/join/JoinHashSupplier.java

```diff
@@ -52,6 +52,31 @@
     private final Optional<JoinFilterFunctionFactory> filterFunctionFactory;
     private final List<JoinFilterFunctionFactory> searchFunctionFactories;
 
+    public static long getEstimatedAdditionalSizeInBytes(
```

Comment:
Could we make it compute all size (not only additional) by passing `PagesIndex#getEstimatedSize()` as param too?

Reply:
Good suggestion, updated